packages/compass-telemetry/src/generic-track.ts

```diff
@@ -33,9 +37,11 @@ export const createTrack = ({
       return;
     }
 
-    if (typeof parameters === 'function') {
+    let parameters: Record<string, unknown> = parametersOrFn;
```

Comment:
this should not work, because at this point we don't know that the parameters are not a function

Reply:
That's true! thanks for catching it! I was using `Omit` on that function type, which just changes the type